### PR TITLE
feat(outputs.prometheus_client): Support vsock listen #6382

### PR DESCRIPTION
**Pull request by jeffrey4l**
This is particial resolved the issue #6382

### CODE REVIEW

File: plugins/outputs/prometheus_client/prometheus_client.go [hunk 1]
```diff
@@ -187,11 +191,38 @@ func (p *PrometheusClient) Init() error {
...
+		return p.listenTCP(p.Listen)
+	}
+	switch strings.ToLower(u.Scheme) {
+	case "tcp", "http":
+		return p.listenTCP(u.Host)
+	case "vsock":
+		return p.listenVsock(u.Host)
 	}
-	return net.Listen("tcp", p.Listen)
+	return nil, errors.New("Unknow scheme")
```
Comment (srebhan) [inline plugins/outputs/prometheus_client/prometheus_client.go:225]: Error messages should start with a lower-case letter... (+typo fix)
```suggestion
	return nil, errors.New("unknown scheme")
```

**Review by srebhan — COMMENTED** [review]
Thank you very much for tackling this old issue @jeffrey4l! Just one minor comment... Furthermore, I wanted to ask if you can also provide a unit-test for the vsock case!?

**Comment by powersj** [comment]
@jeffrey4l,

Thanks for the PR. Sven left a comment above. In addition to that, would you be willing to write a unit test?

Thanks!

**Comment by jeffrey4l** [comment]
sure. will do later

**Comment by jeffrey4l** [comment]
do yo have any idea how to test this? for the vsock, when listen on the host, it only can  be connected from  guest (qemu virtual machine). So i can listen on the vsock, but there is no way test it through real connect.

**Comment by srebhan** [comment]
@jeffrey4l does that somehow work with a container? If so, check using `testcontainers-go` (see e.g. the NTPQ plugin)...

**Comment by jeffrey4l** [comment]
@srebhan no, it only works between host and qemu guest vm. ie listen on the host and connect from the qemu guest vm. or vice versa

**Comment by srebhan** [comment]
@jeffrey4l then I think we are out of luck for a unit-test. Thanks for clarifying! Can you please fix the merge conflicts and fix the two lines I commented on above!? After that I'm fine with the PR...

**Comment by jeffrey4l** [comment]
@srebhan pr is ready for review.

**Review by srebhan — APPROVED** [review]
Thanks for your contribution @jeffrey4l!

**Review by powersj — APPROVED** [review]
Thank you!

**Pull request merged**